Authentication/ConfigBasedAuthenticatorFactory.cs

```diff
@@ -100,10 +100,6 @@ private static Authenticator CreateAuthenticator(Dictionary<string, string> prop
             {
                 authenticator = new BearerTokenAuthenticator(props);
             }
-            else
-            {
-                throw new ArgumentException(string.Format(ErrorMessageAuthTypeUnknown, authType));
-            }
```

Comment:
Is there any reason we are removing this?

Reply:
We have the check for authenticator in `Base service` and we are throwing error there and also the fact that unity is throwing a warning about it because we are not returning all the time.